File: peek_plugin_graphdb/plugin-module/_private/admin/AdminStatusTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {graphDbTuplePrefix} from "../PluginNames";


@addTupleType
export class AdminStatusTuple extends Tuple {
    public static readonly tupleName = graphDbTuplePrefix + "AdminStatusTuple";

    segmentCompilerQueueStatus: boolean;
    segmentCompilerQueueSize: number;
    segmentCompilerQueueProcessedTotal: number;
    segmentCompilerQueueLastError: string;

    constructor() {
        super(AdminStatusTuple.tupleName)
    }
}

File: peek_plugin_graphdb/_private/tuples/AdminStatusTuple.py
from peek_plugin_graphdb._private.PluginNames import graphDbTuplePrefix
from vortex.Tuple import addTupleType, TupleField, Tuple


@addTupleType
class AdminStatusTuple(Tuple):
    __tupleType__ = graphDbTuplePrefix + "AdminStatusTuple"

    segmentCompilerQueueStatus: bool = TupleField(False)
    segmentCompilerQueueSize: int = TupleField(0)
    segmentCompilerQueueProcessedTotal: int = TupleField(0)
    segmentCompilerQueueLastError: str = TupleField()
